cmd/dockerd/hack/malformed_host_override.go

```diff
@@ -38,84 +105,25 @@ func (l *MalformedHostHeaderOverrideConn) Read(b []byte) (n int, err error) {
 		// DefaultMaxHeaderBytes (usually 1 << 20) + 4096.
 		// Here we do the first read which gets us all the http headers to
 		// be inspected and modified below.
-		c, err := l.Conn.Read(b)
+		c, err := l.UnixConn.Read(b)
 		if err != nil {
 			return c, err
 		}
-
-		var (
-			start, end    int
-			firstLineFeed = -1
-			buf           []byte
-		)
-		for i := 0; i <= c-1-7; i++ {
-			if b[i] == '\n' && firstLineFeed == -1 {
-				firstLineFeed = i
-			}
-			if b[i] != '\n' {
-				continue
-			}
-
-			if b[i+1] == '\r' && b[i+2] == '\n' {
-				return c, nil
-			}
-
-			if b[i+1] != 'H' {
-				continue
-			}
-			if b[i+2] != 'o' {
-				continue
-			}
-			if b[i+3] != 's' {
-				continue
-			}
-			if b[i+4] != 't' {
-				continue
-			}
-			if b[i+5] != ':' {
-				continue
-			}
-			if b[i+6] != ' ' {
-				continue
-			}
-			if b[i+7] != '/' {
-				continue
-			}
-			// ensure clients other than the docker clients do not get this hack
-			if i != firstLineFeed {
-				return c, nil
-			}
-			start = i + 7
-			// now find where the value ends
-			for ii, bbb := range b[start:c] {
-				if bbb == '\n' {
-					end = start + ii
-					break
-				}
-			}
-			buf = make([]byte, 0, c+len(closeConnHeader)-(end-start))
-			// strip the value of the host header and
-			// inject `Connection: close` to ensure we don't reuse this connection
-			buf = append(buf, b[:start]...)
-			buf = append(buf, closeConnHeader...)
-			buf = append(buf, b[end:c]...)
-			copy(b, buf)
-			break
-		}
-		if len(buf) == 0 {
-			return c, nil
-		}
-		return len(buf), nil
+		return checkOldHost(c, b)
 	}
-	return l.Conn.Read(b)
+	return l.UnixConn.Read(b)
 }
 
 // Accept makes the listener accepts connections and wraps the connection
 // in a MalformedHostHeaderOverrideConn initializing first to true.
 func (l *MalformedHostHeaderOverride) Accept() (net.Conn, error) {
 	c, err := l.Listener.Accept()
-	if err != nil {
+	if err != nil || c == nil {
 		return c, err
 	}
-	return &MalformedHostHeaderOverrideConn{c, true}, nil
+	unixConn := c.(*net.UnixConn)
```

Comment:
Nit, this is never nil (else it would panic on the type assertion).

Reply:
Thank you for your review. I remove the `c == nil` check and add a `ok` check.